### PR TITLE
Detect when running agent in AKS and set hostname accordingly

### PR DESCRIPTION
**Pull request by jennchenn**
### What does this PR do?

Detect when running in AKS and set the hostname to the node name.

### Motivation

The current default value for `azure_hostname_style` is `os`, which prevents us from using the cloud-provided hostname. This change detects when we are running in AKS and sets the hostname to the kubernetes node name. It also adds this node name as a host alias.

### Describe how to test/QA your changes

1. Deploy the agent to an AKS cluster
2. Exec into an agent pod; verify that the output of `agent hostname` is the same as the node name
3. Check that only one entry in the host list appears for your agent https://dddev.datadoghq.com/infrastructure

### CODE REVIEW

File: releasenotes/notes/use-cloud-provided-hostname-aks-929bfb01e0e336e5.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+enhancements:
+  - |
+    Use cloud-provided hostname as default when running the agent
```
Comment (cswatt) [inline releasenotes/notes/use-cloud-provided-hostname-aks-929bfb01e0e336e5.yaml:11]:
```suggestion
    Use cloud-provided hostname as default when running the Agent
```

**Review by cswatt — APPROVED** [review]

File: pkg/util/cloudproviders/azure/azure.go [hunk 2]
```diff
@@ -53,7 +65,20 @@ var vmIDFetcher = cachedfetch.Fetcher{
 
 // GetHostAliases returns the VM ID from the Azure Metadata api
 func GetHostAliases(ctx context.Context) ([]string, error) {
-	return vmIDFetcher.FetchStringSlice(ctx)
+	aliases := []string{}
+
+	metadata, err := getMetadata(ctx)
+	if err != nil {
+		log.Debugf("Could not get metadata: %s", err)
```
Comment (GustavoCaso) [inline pkg/util/cloudproviders/azure/azure.go:72]: Should return at this point?

File: pkg/util/cloudproviders/azure/azure.go [hunk 3]
```diff
@@ -122,34 +147,52 @@ var instanceMetaFetcher = cachedfetch.Fetcher{
...
 }
 
-func getHostnameWithConfig(ctx context.Context, config config.Config) (string, error) {
-	style := config.GetString(hostnameStyleSetting)
-
-	if style == "os" {
-		return "", fmt.Errorf("azure_hostname_style is set to 'os'")
+func isKubernetesTag(tagsList []map[string]string) bool {
+	for _, tag := range tagsList {
+		if tag["name"] == "aks-managed-orchestrator" && strings.Contains(tag["value"], "Kubernetes") {
```
Comment (GustavoCaso) [inline pkg/util/cloudproviders/azure/azure.go:160]: Could we move the hardcoded string to private constants?

File: pkg/util/cloudproviders/azure/azure.go [hunk 4]
```diff
@@ -162,6 +205,8 @@ func getHostnameWithConfig(ctx context.Context, config config.Config) (string, e
 		name = fmt.Sprintf("%s.%s", metadata.Name, metadata.ResourceGroupName)
 	case "full":
 		name = fmt.Sprintf("%s.%s.%s", metadata.Name, metadata.ResourceGroupName, metadata.SubscriptionID)
+	case "os_computer_name":
+		name = strings.ToLower(metadata.OsProfile.ComputerName)
 	default:
 		return "", fmt.Errorf("invalid azure_hostname_style value: %s", style)
 	}
```
Comment (GustavoCaso) [inline pkg/util/cloudproviders/azure/azure.go:212]: Should this switch case and the style checking be part of a function of the new `metadata` struct? 

```go
metadata.GetFromStyle(style)
```

File: pkg/util/cloudproviders/azure/azure.go [hunk 5]
```diff
@@ -45,15 +75,26 @@ var vmIDFetcher = cachedfetch.Fetcher{
...
 		return []string{res}, nil
 	},
 }
 
 // GetHostAliases returns the VM ID from the Azure Metadata api
 func GetHostAliases(ctx context.Context) ([]string, error) {
-	return vmIDFetcher.FetchStringSlice(ctx)
+	aliases := []string{}
+	vm, err := vmIDFetcher.FetchStringSlice(ctx)
+	aliases = append(aliases, vm...)
```
Comment (GustavoCaso) [inline pkg/util/cloudproviders/azure/azure.go:88]: Should we check the `err` before appending to the `aliases` map?

**Review by GustavoCaso — APPROVED** [review]

**Review by vboulineau — APPROVED** [review]

**Pull request merged**